BeatTogether.DedicatedServer.Kernel/Managers/GameplayManager.cs

```diff
@@ -124,46 +127,49 @@ public async void StartSong(BeatmapIdentifier beatmap, GameplayModifiers modifie
             songReadyCts.CancelAfter((int)(SongLoadTimeLimit * 1000));
             await Task.WhenAll(songReadyTasks);
 
-            // If no players are actually playing
-            if (_playerRegistry.Players.All(player => !player.InGameplay))
-            {
-                _requestReturnToMenuCts.Cancel();
-            }
-
+            // If no players are actually playing, or not all players are not in the lobby(if at least one player is then true)

```

Comment:
prevents the beatmap from playing if a player is not able to load the map, or if someone is stuck in the lobby during map start. returns players to the lobby

Reply:
BUGFIX